src/main/java/org/zendesk/client/v2/Zendesk.java

```diff
@@ -532,6 +534,27 @@ public Iterable<Article> getArticleFromSearch(String locale, String searchTerm,
         return complete(submit(req("GET", tmpl), handleList(Article.class, "results")));
     }
 
+    public MediaRecord getMediaByName(String name) {
+        int apiMaxPageSize = 30;
+        TemplateUri tmpl = tmpl("/guide/medias?filter[name]={name}&page[size]={pageSize}")
+                .set("name", name)
+                .set("pageSize", apiMaxPageSize);
+        MediaResponse response = complete(submit(req("GET", tmpl), handle(MediaResponse.class)));
+        Optional<MediaRecord> optionalMediaRecord = response.getRecords().stream().filter(record -> record.getName().equals(name)).findFirst();
+
+        if (!optionalMediaRecord.isPresent()) {
+            while (response.getMeta().hasMore() && !optionalMediaRecord.isPresent()) {
+                tmpl = tmpl("/guide/medias?filter[name]={name}&page[size]={pageSize}&page[after]={cursor}")
+                        .set("name", name)
+                        .set("pageSize", apiMaxPageSize)
+                        .set("cursor", response.getMeta().getAfterCursor());
+                response = complete(submit(req("GET", tmpl), handle(MediaResponse.class)));
+                optionalMediaRecord = response.getRecords().stream().filter(record -> record.getName().equals(name)).findFirst();
+            }
+        }
```

Comment:
I agree that this needs to be implemented in a client/service class code OR introduce wrapper method like `getAllMedia` with this logic.